crates/red_knot_test/src/parser.rs

```diff
@@ -174,7 +183,7 @@ impl SectionStack {
         popped
     }
 
-    fn parent(&mut self) -> SectionId {
+    fn top(&mut self) -> SectionId {
```

Comment:
I found this function name confusing. It's the current section's ID that's on top of the stack, not the ID of the parent section. When we see a new heading in the parser, this now changes the code to:
```rs
let parent = self.stack.top(); // instead of self.stack.parent()

let section = Section {
    // …
    parent_id: Some(parent),
};

// …

let section_id = self.sections.push(section);
self.stack.push(section_id);

// now `self.stack.top()` refers to the child section
```
but this seems sensible to me. Before we push the new section, the current section is the parent section.